interfaces/builtin/egl_driver_libs.go

```diff
@@ -0,0 +1,156 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/configfiles"
+	"github.com/snapcore/snapd/interfaces/ldconfig"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/snap"
+)
+
+const eglDriverLibsSummary = `allows exposing EGL driver libraries to the system`
+
+// Plugs only supported for the system on classic for the moment
+const eglDriverLibsBaseDeclarationPlugs = `
+  egl-driver-libs:
+    allow-installation:
+      plug-snap-type:
+        - core
+      on-classic: true
```

Comment:
I'm not entirely sure of the need/effect of this given that the plug is a system plug only atm, until we add app to the types it's not clear it's needed but we probably shouldn't add app until things work on core too

Reply:
I think it's mostly a bit confusing given that we have the control on the interface if the plug exists at all, and actually we exepect the non-system plugs if any to be used on UC

Reply:
I have now changed this to `allow-installation: false` and added a comment to clarify this.